### PR TITLE
chore: update mergify and dependabot

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: NA

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

**Review by yaruwangway — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ updates:
     directory: "/"
     schedule:
       interval: daily
-    target-branch: "release/v10.0.x"
+    target-branch: "release/v11.x"
```
Comment (p-offtermatt) [inline .github/dependabot.yml:44]: Should we add a comment that this is the last 3 releases, with a reference to e.g. https://github.com/cosmos/gaia/blob/main/RELEASING.md#major-release-maintenance?

File: .mergify.yml [hunk 2]
```diff
@@ -24,29 +24,29 @@ pull_request_rules:
           {{ title }} (#{{ number }})
           {{ body }}
 
-  - name: Backport patches to the release/v10.0.x branch
+  - name: Backport patches to the release/v11.x branch
```
Comment (p-offtermatt) [inline .mergify.yml:27]: Same as above, maybe we should put a comment that this is supposed to be the last 3 releases included here

**Review by p-offtermatt — APPROVED** [review]
LGTM, just a comment. I think it probably shouldn't be addressed in this PR anyways, but I'll let you decide how to proceed/if you think it's worth doing

File: .github/dependabot.yml (hunk 1 above)
Comment (mpoke) [reply .github/dependabot.yml:44]: We should keep it as at the moment the latest two _in-production_ releases are v11 and v12.

**Pull request merged**